src/nbl/ui/CWindowManagerWin32.cpp

```diff
@@ -29,59 +29,48 @@ IWindowManager::SDisplayInfo CWindowManagerWin32::getPrimaryDisplayInfo() const
 static inline DWORD getWindowStyle(const core::bitflag<IWindow::E_CREATE_FLAGS> flags)
 {
 	DWORD style = WS_POPUP;
+	// These are always set by GLFW
+	style |= (WS_CLIPCHILDREN | WS_CLIPSIBLINGS);
 
-	if (!flags.hasFlags(IWindow::ECF_FULLSCREEN))
+	if (!flags.hasFlags(IWindow::ECF_BORDERLESS))
 	{
-		if (!flags.hasFlags(IWindow::ECF_BORDERLESS))
-		{
-			style |= WS_BORDER;
-			style |= (WS_SYSMENU | WS_CAPTION);
-		}
-		// ? not sure about those below
-		style |= WS_CLIPCHILDREN;
-		style |= WS_CLIPSIBLINGS;
+		style |= WS_BORDER | WS_CAPTION | WS_SYSMENU;
 	}
-	if (flags.hasFlags(IWindow::ECF_MINIMIZED))
+	if (flags.hasFlags(IWindow::ECF_UI_RESIZABLE))
 	{
-		style |= WS_MINIMIZE;
+		style |= WS_SIZEBOX;
 	}
 	if (flags.hasFlags(IWindow::ECF_MAXIMIZED))
 	{
 		style |= WS_MAXIMIZE;
 	}
-	if (flags.hasFlags(IWindow::ECF_ALWAYS_ON_TOP))
-	{
-		style |= WS_EX_TOPMOST;
-	}
-	if (!flags.hasFlags(IWindow::ECF_HIDDEN))
+	if (flags.hasFlags(IWindow::ECF_MINIMIZED))
 	{
-		style |= WS_VISIBLE;
+		style |= WS_MINIMIZE;
 	}
-	style |= WS_OVERLAPPEDWINDOW;
-	if (!flags.hasFlags(IWindow::ECF_CAN_RESIZE))
+	if (flags.hasFlags(IWindow::ECF_CAN_MAXIMIZE))
 	{
-		style &= ~WS_SIZEBOX;
+		style |= WS_MAXIMIZEBOX;
 	}
-	if (!flags.hasFlags(IWindow::ECF_CAN_MAXIMIZE))
+	if (flags.hasFlags(IWindow::ECF_CAN_MINIMIZE))
 	{
-		style &= ~WS_MAXIMIZEBOX;
+		style |= WS_MINIMIZEBOX;
 	}
-	if (!flags.hasFlags(IWindow::ECF_CAN_MINIMIZE))
+	if (!flags.hasFlags(IWindow::ECF_HIDDEN))
 	{
-		style &= ~WS_MINIMIZEBOX;
+		style |= WS_VISIBLE;
 	}
 
 	return style;
 }
 
 core::smart_refctd_ptr<IWindow> CWindowManagerWin32::createWindow(IWindow::SCreationParams&& creationParams)
 {
-	// this could be common to all `createWindow` impl
-	if (creationParams.flags.hasFlags(IWindow::ECF_CAN_RESIZE) || creationParams.flags.hasFlags(IWindow::ECF_CAN_MAXIMIZE))
-		creationParams.flags |= IWindow::ECF_RESIZABLE;
 	// win32 minimize is weird, its a resize to 0,0
-	if (creationParams.flags.hasFlags(IWindow::ECF_CAN_MINIMIZE))
-		creationParams.flags |= IWindow::ECF_CAN_RESIZE;
+	if (creationParams.flags.hasFlags(IWindow::ECF_CAN_MAXIMIZE) || creationParams.flags.hasFlags(IWindow::ECF_CAN_MINIMIZE))
+		creationParams.flags |= IWindow::ECF_UI_RESIZABLE;
+	if (creationParams.flags.hasFlags(IWindow::ECF_FULLSCREEN))
```

Comment:
Hmm I wonder why games have full screen vs windowed borderless fullscreen, do we need to consider this as well? Why are you adding borderless flag when you detect fullscreen?

Reply:
i did some researching and apparently in windows, the only difference between *fullscreen* and *windowed borderless fullscreen* is that the former has the `HWND_TOPMOST` flag. so:
- fullscreen: `borderless + always-on-top + monitor sized`
- windowed borderless fullscreen: `borderless + monitor sized`